### ISSUE TITLE
Make use of new Broadcast

### ISSUE THREAD

**Issue by Evizero** [JuliaML/LossFunctions.jl#61]
So this is more of a discussion. I though naively it may be nice to make use of the new broadcast syntax (e.g. `sin.([1,2,3])`) for our loss functions.

It is not hard to make work. We would just have to define some dummy methods. That said, our manual implementation seems superior at the moment

```julia
Base.getindex(l::Loss, idx) = l
Base.size(::Loss) = (1,)
```

After compiling this is the result

```julia
julia> @time value.(L2DistLoss(), [11,-3,7,14], [1,2,3,4])
  0.000522 seconds (108 allocations: 3.359 KB)
4-element Array{Int64,1}:
 100
  25
  16
 100
```

As comparison this is our [manual implementation](https://github.com/JuliaML/LossFunctions.jl/blob/master/src/supervised/supervised.jl#L86-L102)

```julia
julia> @time value(L2DistLoss(), [11,-3,7,14], [1,2,3,4])
  0.000014 seconds (7 allocations: 496 bytes)
4-element Array{Int64,1}:
 100
  25
  16
 100
```

Thoughts?

**Comment by ChrisRackauckas** [comment]
The amount of allocations shouldn't be that different. Are you sure you aren't missing a function?

**Comment by Evizero** [comment]
I am sure of nothing here. This was more of a quick tests to see how it would behave

**Comment by Evizero** [comment]
@StefanKarpinski sorry to ping you out of the blue, but do you have any pointers or hints on this? I would ideally like to move away from providing custom vectorized implementations.

**Comment by Evizero** [comment]
Putting it into a function helped a little.

```julia
fun_new(t,y) = value.(L2DistLoss(), t, y)
```

```julia
# after warmup
@time fun_new([11,-3,7,14], [1,2,3,4])
```

```
julia> @time fun_new([11,-3,7,14], [1,2,3,4])
  0.000236 seconds (76 allocations: 2.266 KB)
4-element Array{Int64,1}:
 100
  25
  16
 100
```

But even with large vectors I still have a 40x speed difference according to `@benchmark` 

```
julia> t = rand(1000); y = rand(1000);
julia> @benchmark value(L2DistLoss(), t, y)
BenchmarkTools.Trial: 
  memory estimate:  7.94 kb
  allocs estimate:  1
  --------------
  minimum time:     851.034 ns (0.00% GC)
  median time:      1.689 μs (0.00% GC)
  mean time:        2.707 μs (29.54% GC)
  maximum time:     93.844 μs (96.84% GC)
  --------------
  samples:          10000
  evals/sample:     29
  time tolerance:   5.00%
  memory tolerance: 1.00%
```

vs

```
julia> @benchmark fun_new(t, y)
BenchmarkTools.Trial: 
  memory estimate:  10.73 kb
  allocs estimate:  99
  --------------
  minimum time:     65.694 μs (0.00% GC)
  median time:      77.117 μs (0.00% GC)
  mean time:        80.199 μs (0.38% GC)
  maximum time:     3.277 ms (91.87% GC)
  --------------
  samples:          10000
  evals/sample:     1
  time tolerance:   5.00%
  memory tolerance: 1.00%
```

**Comment by Evizero** [comment]
implemented in #68
